pkg/controller/replica.go

```diff
@@ -133,7 +133,10 @@ func (r *ReconcileReplica) Reconcile(ctx context.Context, request reconcile.Requ
 func (r *ReconcileReplica) triggerGarbageCollection(volumeName string) {
 	emptyCtx := context.TODO()
 	deletionCtx, cancelFunc := context.WithCancel(emptyCtx)
-	_, _ = r.cleanUpMap.LoadOrStore(volumeName, cancelFunc)
+	if _, ok := r.cleanUpMap.LoadOrStore(volumeName, cancelFunc); ok {
+		klog.Infof("There already is a scheduled garbage collection for AzVolume (%s)")
+		return
```

Comment:
Need invoke `cancelFunc` if here since we're not using it and need to clean up the resources the context holds.